In [0]:
from pyspark.sql.functions import *
from pyspark.sql.types import *

In [0]:
#Učitamo transformisane podatke

file_path = "/tmp/happiness/"

hapDF = spark.read.csv(file_path, header=True, inferSchema=True)

hapDF.display()

rank country score 1 finland 7.769 2 denmark 7.6 3 norway 7.554 4 iceland 7.494 5 netherlands 7.488 6 switzerland 7.48 7 sweden 7.343 8 new zealand 7.307 9 canada 7.278 10 austria 7.246 11 australia 7.228 12 costa rica 7.167 13 israel 7.139 14 luxembourg 7.09 15 united kingdom 7.054 16 ireland 7.021 17 germany 6.985 18 belgium 6.923 19 united states 6.892 20 czech republic 6.852 21 united arab emirates 6.825 22 malta 6.726 23 mexico 6.595 24 france 6.592 25 taiwan 6.446 26 chile 6.444 27 guatemala 6.436 28 saudi arabia 6.375 29 qatar 6.374 30 spain 6.354 31 panama 6.321 32 brazil 6.3 33 uruguay 6.293 34 singapore 6.262 35 el salvador 6.253 36 italy 6.223 37 bahrain 6.199 38 slovakia 6.198 39 trinidad & tobago 6.192 40 poland 6.182 41 uzbekistan 6.174 42 lithuania 6.149 43 colombia 6.125 44 slovenia 6.118 45 nicaragua 6.105 46 kosovo 6.1 47 argentina 6.086 48 romania 6.07 49 cyprus 6.046 50 ecuador 6.028 51 kuwait 6.021 52 thailand 6.008 53 latvia 5.94 54 south korea 5.895 55 estonia 5.893 56 jamaica 5.89 57 mauritius 5.888 58 japan 5.886 59 honduras 5.86 60 kazakhstan 5.809 61 bolivia 5.779 62 hungary 5.758 63 paraguay 5.743 64 northern cyprus 5.718 65 peru 5.697 66 portugal 5.693 67 pakistan 5.653 68 russia 5.648 69 philippines 5.631 70 serbia 5.603 71 moldova 5.529 72 libya 5.525 73 montenegro 5.523 74 tajikistan 5.467 75 croatia 5.432 76 hong kong 5.43 77 dominican republic 5.425 78 bosnia and herzegovina 5.386 79 turkey 5.373 80 malaysia 5.339 81 belarus 5.323 82 greece 5.287 83 mongolia 5.285 84 north macedonia 5.274 85 nigeria 5.265 86 kyrgyzstan 5.261 87 turkmenistan 5.247 88 algeria 5.211 89 morocco 5.208 90 azerbaijan 5.208 91 lebanon 5.197 92 indonesia 5.192 93 china 5.191 94 vietnam 5.175 95 bhutan 5.082 96 cameroon 5.044 97 bulgaria 5.011 98 ghana 4.996 99 ivory coast 4.944 100 nepal 4.913 101 jordan 4.906 102 benin 4.883 103 congo, rep. 4.812 104 gabon 4.799 105 laos 4.796 106 south africa 4.722 107 albania 4.719 108 venezuela 4.707 109 cambodia 4.7 110 palestinian territories 4.696 111 senegal 4.681 112 somalia 4.668 113 namibia 4.639 114 niger 4.628 115 burkina faso 4.587 116 armenia 4.559 117 iran 4.548 118 guinea 4.534 119 georgia 4.519 120 gambia 4.516 121 kenya 4.509 122 mauritania 4.49 123 mozambique 4.466 124 tunisia 4.461 125 bangladesh 4.456 126 iraq 4.437 127 congo, dem. rep. 4.418 128 mali 4.39 129 sierra leone 4.374 130 sri lanka 4.366 131 myanmar 4.36 132 chad 4.35 133 ukraine 4.332 134 ethiopia 4.286 135 swaziland 4.212 136 uganda 4.189 137 egypt 4.166 138 zambia 4.107 139 togo 4.085 140 india 4.015 141 liberia 3.975 142 comoros 3.973 143 madagascar 3.933 144 lesotho 3.802 145 burundi 3.775 146 zimbabwe 3.663 147 haiti 3.597 148 botswana 3.488 149 syria 3.462 150 malawi 3.41 151 yemen 3.38 152 rwanda 3.334 153 tanzania 3.231 154 afghanistan 3.203 155 central african republic 3.083 156 south sudan 2.853

__Prikaz mape sveta sa koeficijentom sreće__

In [0]:
newHapDF = hapDF.join(suDF, hapDF.country == suDF.country, "inner").drop(suDF.country).drop("year")
display(newHapDF)

rank country score iso3c suicide_rate gdp 154 afghanistan 3.203 AFG 4.1 1.87994507427823E10 107 albania 4.719 ALB 4.3 1.54002428748812E10 88 algeria 5.211 DZA 2.5 1.7176740374819E11 47 argentina 6.086 ARG 8.4 4.51932356085842E11 116 armenia 4.559 ARM 3.3 1.36192913612814E10 11 australia 7.228 AUS 12.5 1.39195251037048E12 10 austria 7.246 AUT 14.6 4.4501187270447E11 90 azerbaijan 5.208 AZE 4.1 4.81742352941176E10 37 bahrain 6.199 BHR 8.9 3.8652579787234E10 125 bangladesh 4.456 BGD 3.7 3.0257132044575E11 81 belarus 5.323 BLR 21.2 6.44096471938044E10 18 belgium 6.923 BEL 18.3 5.35288715239999E11 102 benin 4.883 BEN 7.8 1.43916866327821E10 95 bhutan 5.082 BTN 4.6 2.53565706896575E9 61 bolivia 5.779 BOL 6.2 4.08953228437851E10 78 bosnia and herzegovina 5.386 BIH 10.9 2.02013232600733E10 148 botswana 3.488 BWA 16.1 1.65937206556402E10 32 brazil 6.3 BRA 6.9 1.87782427372078E12 97 bulgaria 5.011 BGR 9.7 6.89154161419576E10 115 burkina faso 4.587 BFA 7.5 1.61781620300694E10 145 burundi 3.775 BDI 6.2 2.63143436322693E9 109 cambodia 4.7 KHM 4.9 2.7089389786979E10 96 cameroon 5.044 CMR 9.0 3.96709773327348E10 9 canada 7.278 CAN 11.8 1.74201504548231E12 155 central african republic 3.083 CAF 12.3 2.22030736869593E9 132 chad 4.35 TCD 6.4 1.13149513427807E10 26 chile 6.444 CHL 9.0 2.79385487344492E11 93 china 5.191 CHN 8.1 1.4279937467431E13 43 colombia 6.125 COL 3.9 3.23429888934257E11 142 comoros 3.973 COM 5.4 1.19255973937288E9 127 congo, dem. rep. 4.418 COD 6.7 5.0400747050496E10 103 congo, rep. 4.812 COG 6.5 1.27503390229512E10 12 costa rica 7.167 CRI 8.1 6.4072870931603E10 75 croatia 5.432 HRV 16.4 6.22462063405469E10 49 cyprus 6.046 CYP 3.6 2.57583577745438E10 20 czech republic 6.852 CZE 12.2 2.52498032247163E11 2 denmark 7.6 DNK 10.7 3.47561349210979E11 77 dominican republic 5.425 DOM 4.9 8.89412997335018E10 50 ecuador 6.028 ECU 7.6 1.08108009E11 137 egypt 4.166 EGY 3.0 3.03080865603645E11 35 el salvador 6.253 SLV 6.1 2.689666E10 55 estonia 5.893 EST 14.9 3.10455917534901E10 134 ethiopia 4.286 ETH 5.4 9.59125906281412E10 1 finland 7.769 FIN 15.3 2.6850820012549E11 24 france 6.592 FRA 13.8 2.72887024670588E12 104 gabon 4.799 GAB 8.4 1.68744058397948E10 119 georgia 4.519 GEO 9.2 1.74772555622583E10 17 germany 6.985 DEU 12.3 3.88832678862744E12 98 ghana 4.996 GHA 6.6 6.83375378157703E10 82 greece 5.287 GRC 5.1 2.05144152830847E11 27 guatemala 6.436 GTM 5.9 7.70200152013201E10 118 guinea 4.534 GIN 7.0 1.35138092582131E10 147 haiti 3.597 HTI 9.6 1.47858393829002E10 59 honduras 5.86 HND 2.1 2.49155228491544E10 62 hungary 5.758 HUN 16.6 1.6352649143329E11 4 iceland 7.494 ISL 11.9 2.48577404450401E10 140 india 4.015 IND 12.7 2.87050409671777E12 92 indonesia 5.192 IDN 2.4 1.11909125907462E12 117 iran 4.548 IRN 5.2 2.58245497664394E11 126 iraq 4.437 IRQ 3.6 2.35097182233503E11 16 ireland 7.021 IRL 9.6 3.99122063504148E11 13 israel 7.139 ISR 5.3 3.9793459695256E11 36 italy 6.223 ITA 6.7 2.00938386730745E12 56 jamaica 5.89 JAM 2.4 1.58307685498916E10 58 japan 5.886 JPN 15.3 5.14878194847817E12 101 jordan 4.906 JOR 1.6 4.4502895915493E10 60 kazakhstan 5.809 KAZ 17.6 1.81667190075541E11 121 kenya 4.509 KEN 6.1 1.00555485831944E11 51 kuwait 6.021 KWT 2.9 1.36196760210804E11 53 latvia 5.94 LVA 20.1 3.43087838253019E10 91 lebanon 5.197 LBN 2.8 5.19537445074091E10 144 lesotho 3.802 LSO 72.4 2.36621306857507E9 141 liberia 3.975 LBR 4.5 3.3195965E9 72 libya 5.525 LBY 4.5 5.20911522283425E10 42 lithuania 6.149 LTU 26.1 5.46973790173332E10 14 luxembourg 7.09 LUX 11.3 7.01957154955136E10 143 madagascar 3.933 MDG 5.5 1.410466451487E10 150 malawi 3.41 MWI 5.4 1.1025357837607E10 80 malaysia 5.339 MYS 5.7 3.65276282438141E11 128 mali 4.39 MLI 4.1 1.72802511939527E10 22 malta 6.726 MLT 6.1 1.52157143090974E10 122 mauritania 4.49 MRT 3.1 7.88966245216785E9 57 mauritius 5.888 MUS 9.5 1.4045808843221E10 23 mexico 6.595 MEX 5.3 1.26943393280591E12 71 moldova 5.529 MDA 14.7 1.19702339398808E10 83 mongolia 5.285 MNG 17.9 1.42063590068095E10 73 montenegro 5.523 MNE 21

In [0]:
#Učitamo transformisane podatke
file_path = "/tmp/alcohol/"

alcDF = spark.read.csv(file_path, header=True, inferSchema=True)
alcDF.display()

country both male female afghanistan 0.013 0.022 0.003 albania 6.82 11.02 2.53 algeria 0.63 1.09 0.17 andorra 12.26 18.58 5.74 angola 7.83 12.39 3.48 antigua and barbuda 9.4 15.09 4.25 argentina 9.45 14.66 4.64 armenia 4.67 7.7 2.14 australia 10.36 15.72 5.11 austria 11.93 18.42 5.76 azerbaijan 0.98 1.67 0.32 bahamas 4.8 7.66 2.16 bahrain 1.13 1.54 0.32 bangladesh 0.0 0.0 0.0 barbados 10.37 16.95 4.35 belarus 10.96 17.93 5.13 belgium 10.8 16.7 5.13 belize 6.37 10.18 2.65 benin 2.23 3.59 0.9 bhutan 0.21 0.33 0.077 bolivia 3.91 6.0 1.84 bosnia and herzegovina 7.81 12.95 2.96 botswana 6.61 11.1 2.61 brazil 7.32 11.65 3.22 brunei 0.46 0.68 0.21 bulgaria 12.46 19.86 5.59 burkina faso 11.05 18.5 3.85 burundi 7.45 12.22 2.89 cambodia 7.84 12.9 3.21 cameroon 5.52 8.76 2.32 canada 8.81 13.73 4.01 central african republic 1.71 2.76 0.71 chad 1.26 2.07 0.46 chile 8.95 13.74 4.36 china 6.04 9.37 2.6 colombia 5.45 8.84 2.27 comoros 1.15 2.0 0.3 cook islands 10.94 17.95 4.36 costa rica 4.07 6.49 1.68 croatia 8.73 13.96 3.99 cuba 6.26 9.97 2.64 cyprus 10.8 16.92 4.76 denmark 10.13 15.58 4.8 djibouti 0.36 0.6 0.093 dominica 7.23 11.53 2.9 dominican republic 6.68 10.62 2.8 dr congo 1.1 1.76 0.46 ecuador 3.27 5.01 1.56 egypt 0.14 0.24 0.037 el salvador 4.09 6.91 1.75 equatorial guinea 6.92 9.63 3.12 eritrea 2.11 3.52 0.73 estonia 10.75 17.35 5.06 eswatini 8.78 14.78 3.26 ethiopia 2.2 3.66 0.78 fiji 3.68 6.08 1.23 finland 10.65 16.56 4.99 france 12.23 19.21 5.83 gabon 8.08 12.24 3.7 gambia 3.36 5.73 1.09 georgia 9.47 15.61 4.11 germany 12.79 19.71 6.15 ghana 2.78 4.37 1.16 greece 10.5 16.72 4.6 grenada 8.96 14.25 3.64 guatemala 1.63 2.63 0.69 guinea 1.09 1.94 0.36 guinea-bissau 5.54 9.32 2.09 guyana 5.33 8.57 2.1 haiti 3.02 4.96 1.19 honduras 3.93 6.34 1.57 hungary 11.07 17.66 5.24 iceland 9.21 14.03 4.39 india 5.61 9.06 1.89 indonesia 0.22 0.37 0.063 iran 1.01 1.76 0.25 iraq 0.37 0.64 0.1 ireland 12.75 19.38 6.34 israel 4.38 6.84 2.02 italy 8.01 12.74 3.61 ivory coast 3.04 4.83 1.19 jamaica 4.16 6.71 1.69 japan 10.09 15.58 4.93 jordan 0.52 0.89 0.14 kazakhstan 5.0 8.33 2.02 kenya 2.15 3.51 0.83 kiribati 2.32 3.97 0.8 kuwait 0.0 0.0 0.0 kyrgyzstan 4.85 8.38 1.51 laos 12.08 19.11 5.09 latvia 13.19 21.61 6.31 lebanon 1.53 2.58 0.49 lesotho 5.08 8.65 1.67 liberia 5.37 8.58 2.2 libya 0.027 0.047 0.007 lithuania 12.78 20.6 6.3 luxembourg 12.45 18.74 6.08 madagascar 1.98 3.22 0.77 malawi 4.08 6.76 1.56 malaysia 0.93 1.48 0.35 maldives 2.78 3.95 0.57 mali 1.31 2.26 0.39 malta 8.27 12.88 3.69 mauritania 0.0 0.0 0.0 mauritius 4.78 7.67 2.0 mexico 5.05 8.13 2.18 micronesia 2.51 4.14 0.85 moldova 12.85 20.61 5.89 mongolia 5.9 9.23 2.72 montenegro 12.24 19.51 5.3 morocco 0.49 0.87 0.13 mozambique 2.69 4.53 1.04 myanmar 2.14 3.57 0.85 namibia 3.15 5.3 1.2 nauru 4.18 7.07 1.46 nepal 0.62 1.1 0.25 netherlands 9.67 14.9 4.54 new zealand 10.69 16.46 5.23 nicaragua 5.07 8.31 2.02 niger 0.52 0.89 0.15 nigeria 6.19 9.84 2.49 niue 9.93 16.04 3.77 north korea 4.16 6.89 1.61 north macedonia 6.43 10.36 2.54 norway 7.14 10.8 3.45 oman 0.92 1.22 0.23 pakistan 0.31 0.52 0.095 panama 7.77 12.23 3.36 papua new guinea 2.12 3.49 0.72 paraguay 7.01 10.96 2.94 peru 6.78 10.41 3.24 philippines 7.02 11.22 2.87 poland 11.89 18.65 5.65 portugal 12.09 19.51 5.59 qatar 1.51 1.81 0.38 romania 12.34 19.5 5.68 russia 10.5 17.99 4.31 rwanda 7.96 13.11 3.11 saint kitts and nevis 6.34 10.01 2.78 saint lucia 9.59 15.45 3.97 saint vincent and the grenadines 7.15 11.29 2.87 samoa 2.82 4.58 0.93 sao tome and principe 5.8 9.15 2.49 saudi arabia 0.0 0.0 0.0 senegal 0.73 1.3 0.22 serbia 8.85 14.09 3.93 seychelles 8.84 13.41 4.02 sierra leone 5.31 9.19 1.47 singapore 2.0 2.92 0.99 slovakia 11.06 17.26 5.3 slovenia 12.11 18.68 5.67 solomon islands 1.74 2.87 0.59 somalia 0.0 0.0 0.0 south africa 9.45 15.74 3.46 south korea 8.45 12.76 4.17 spain 12.67 19.94 5.76 sri lanka 2.87 4.75 1.19 suriname 7.4 11.74 3.1 sweden 9.04 13.85 4.27 switzerland 11.23 17.18 5.45 syria 0.19 0.32 0.047 tajikistan 0.88 1.

__Prikaz mape sveta sa konzumpcijom alkohola za oba pola__

In [0]:
newAlcDF = alcDF.join(suDF, alcDF.country == suDF.country, "inner").drop(suDF.country).drop("year")
display(newAlcDF)


country both male female iso3c suicide_rate gdp afghanistan 0.013 0.022 0.003 AFG 4.1 1.87994507427823E10 albania 6.82 11.02 2.53 ALB 4.3 1.54002428748812E10 algeria 0.63 1.09 0.17 DZA 2.5 1.7176740374819E11 angola 7.83 12.39 3.48 AGO 6.1 8.94171903412533E10 antigua and barbuda 9.4 15.09 4.25 ATG 0.4 1.68753333333333E9 argentina 9.45 14.66 4.64 ARG 8.4 4.51932356085842E11 armenia 4.67 7.7 2.14 ARM 3.3 1.36192913612814E10 australia 10.36 15.72 5.11 AUS 12.5 1.39195251037048E12 austria 11.93 18.42 5.76 AUT 14.6 4.4501187270447E11 azerbaijan 0.98 1.67 0.32 AZE 4.1 4.81742352941176E10 bahrain 1.13 1.54 0.32 BHR 8.9 3.8652579787234E10 bangladesh 0.0 0.0 0.0 BGD 3.7 3.0257132044575E11 barbados 10.37 16.95 4.35 BRB 0.6 5.209E9 belarus 10.96 17.93 5.13 BLR 21.2 6.44096471938044E10 belgium 10.8 16.7 5.13 BEL 18.3 5.35288715239999E11 belize 6.37 10.18 2.65 BLZ 7.1 1.9825185406134E9 benin 2.23 3.59 0.9 BEN 7.8 1.43916866327821E10 bhutan 0.21 0.33 0.077 BTN 4.6 2.53565706896575E9 bolivia 3.91 6.0 1.84 BOL 6.2 4.08953228437851E10 bosnia and herzegovina 7.81 12.95 2.96 BIH 10.9 2.02013232600733E10 botswana 6.61 11.1 2.61 BWA 16.1 1.65937206556402E10 brazil 7.32 11.65 3.22 BRA 6.9 1.87782427372078E12 bulgaria 12.46 19.86 5.59 BGR 9.7 6.89154161419576E10 burkina faso 11.05 18.5 3.85 BFA 7.5 1.61781620300694E10 burundi 7.45 12.22 2.89 BDI 6.2 2.63143436322693E9 cambodia 7.84 12.9 3.21 KHM 4.9 2.7089389786979E10 cameroon 5.52 8.76 2.32 CMR 9.0 3.96709773327348E10 canada 8.81 13.73 4.01 CAN 11.8 1.74201504548231E12 central african republic 1.71 2.76 0.71 CAF 12.3 2.22030736869593E9 chad 1.26 2.07 0.46 TCD 6.4 1.13149513427807E10 chile 8.95 13.74 4.36 CHL 9.0 2.79385487344492E11 china 6.04 9.37 2.6 CHN 8.1 1.4279937467431E13 colombia 5.45 8.84 2.27 COL 3.9 3.23429888934257E11 comoros 1.15 2.0 0.3 COM 5.4 1.19255973937288E9 costa rica 4.07 6.49 1.68 CRI 8.1 6.4072870931603E10 croatia 8.73 13.96 3.99 HRV 16.4 6.22462063405469E10 cuba 6.26 9.97 2.64 CUB 14.5 1.03428E11 cyprus 10.8 16.92 4.76 CYP 3.6 2.57583577745438E10 denmark 10.13 15.58 4.8 DNK 10.7 3.47561349210979E11 djibouti 0.36 0.6 0.093 DJI 9.6 3.32461554909099E9 dominican republic 6.68 10.62 2.8 DOM 4.9 8.89412997335018E10 ecuador 3.27 5.01 1.56 ECU 7.6 1.08108009E11 egypt 0.14 0.24 0.037 EGY 3.0 3.03080865603645E11 el salvador 4.09 6.91 1.75 SLV 6.1 2.689666E10 equatorial guinea 6.92 9.63 3.12 GNQ 7.9 1.14172780449859E10 estonia 10.75 17.35 5.06 EST 14.9 3.10455917534901E10 eswatini 8.78 14.78 3.26 SWZ 29.4 4.49567948096886E9 ethiopia 2.2 3.66 0.78 ETH 5.4 9.59125906281412E10 fiji 3.68 6.08 1.23 FJI 9.0 5.49626476637285E9 finland 10.65 16.56 4.99 FIN 15.3 2.6850820012549E11 france 12.23 19.21 5.83 FRA 13.8 2.72887024670588E12 gabon 8.08 12.24 3.7 GAB 8.4 1.68744058397948E10 georgia 9.47 15.61 4.11 GEO 9.2 1.74772555622583E10 germany 12.79 19.71 6.15 DEU 12.3 3.88832678862744E12 ghana 2.78 4.37 1.16 GHA 6.6 6.83375378157703E10 greece 10.5 16.72 4.6 GRC 5.1 2.05144152830847E11 grenada 8.96 14.25 3.64 GRD 0.7 1.21269440740741E9 guatemala 1.63 2.63 0.69 GTM 5.9 7.70200152013201E10 guinea 1.09 1.94 0.36 GIN 7.0 1.35138092582131E10 guinea-bissau 5.54 9.32 2.09 GNB 7.0 1.43963844338133E9 guyana 5.33 8.57 2.1 GUY 40.3 5.17376019184652E9 haiti 3.02 4.96 1.19 HTI 9.6 1.47858393829002E10 honduras 3.93 6.34 1.57 HND 2.1 2.49155228491544E10 hungary 11.07 17.66 5.24 HUN 16.6 1.6352649143329E11 iceland 9.21 14.03 4.39 ISL 11.9 2.48577404450401E10 india 5.61 9.06 1.89 IND 12.7 2.87050409671777E12 indonesia 0.22 0.37 0.063 IDN 2.4 1.11909125907462E12 iran 1.01 1.76 0.25 IRN 5.2 2.58245497664394E11 iraq 0.37 0.64 0.1 IRQ 3.6 2.35097182233503E11 ireland 12.75 19.38 6.34 IRL 9.6 3.99122063504148E11 israel 4.38 6.84 2.02 ISR 5.3 3.9793459695256E11 italy 8.01 12.74 3.61 ITA 6.7 2.00938386730745E12 jamaica 4.16 6.71 1.69 JAM 2.4 1.58307685498916E10 japan 10.09 15.58 4.93 JPN 15.3 5.14878194847817E12 jordan 0.52 0.89 0.14 JOR 1.6 4.4502895915493E10 kazakhstan 5.0 8.33 2.02 KAZ 17.6 1.81667190075541E11 kenya 2.15 

__Prikaz mape sveta sa stopom samoubistva__

In [0]:
#Učitamo transformisane podatke
file_path = "/tmp/suicide/"

suDF = spark.read.csv(file_path, header=True, inferSchema=True)
suDF.display() 

country iso3c year suicide_rate gdp afghanistan AFG 2019 4.1 1.87994507427823E10 africa eastern and southern AFE 2019 8.03629338807931 1.00990985542184E12 africa western and central AFW 2019 5.40949890228626 7.94572474605872E11 albania ALB 2019 4.3 1.54002428748812E10 algeria DZA 2019 2.5 1.7176740374819E11 angola AGO 2019 6.1 8.94171903412533E10 antigua and barbuda ATG 2019 0.4 1.68753333333333E9 arab world ARB 2019 4.18032530134748 2.77646869232296E12 argentina ARG 2019 8.4 4.51932356085842E11 armenia ARM 2019 3.3 1.36192913612814E10 australia AUS 2019 12.5 1.39195251037048E12 austria AUT 2019 14.6 4.4501187270447E11 azerbaijan AZE 2019 4.1 4.81742352941176E10 bahamas, the BHS 2019 3.5 1.31644E10 bahrain BHR 2019 8.9 3.8652579787234E10 bangladesh BGD 2019 3.7 3.0257132044575E11 barbados BRB 2019 0.6 5.209E9 belarus BLR 2019 21.2 6.44096471938044E10 belgium BEL 2019 18.3 5.35288715239999E11 belize BLZ 2019 7.1 1.9825185406134E9 benin BEN 2019 7.8 1.43916866327821E10 bhutan BTN 2019 4.6 2.53565706896575E9 bolivia BOL 2019 6.2 4.08953228437851E10 bosnia and herzegovina BIH 2019 10.9 2.02013232600733E10 botswana BWA 2019 16.1 1.65937206556402E10 brazil BRA 2019 6.9 1.87782427372078E12 brunei darussalam BRN 2019 2.7 1.34694229585105E10 bulgaria BGR 2019 9.7 6.89154161419576E10 burkina faso BFA 2019 7.5 1.61781620300694E10 burundi BDI 2019 6.2 2.63143436322693E9 cabo verde CPV 2019 12.9 1.98184574070615E9 cambodia KHM 2019 4.9 2.7089389786979E10 cameroon CMR 2019 9.0 3.96709773327348E10 canada CAN 2019 11.8 1.74201504548231E12 caribbean small states CSS 2019 9.82628287062887 7.61736694811443E10 central african republic CAF 2019 12.3 2.22030736869593E9 central europe and the baltics CEB 2019 12.5212571972586 1.67386331097542E12 chad TCD 2019 6.4 1.13149513427807E10 chile CHL 2019 9.0 2.79385487344492E11 china CHN 2019 8.1 1.4279937467431E13 colombia COL 2019 3.9 3.23429888934257E11 comoros COM 2019 5.4 1.19255973937288E9 congo, dem. rep. COD 2019 6.7 5.0400747050496E10 congo, rep. COG 2019 6.5 1.27503390229512E10 costa rica CRI 2019 8.1 6.4072870931603E10 cote d'ivoire CIV 2019 8.9 5.85394249297248E10 croatia HRV 2019 16.4 6.22462063405469E10 cuba CUB 2019 14.5 1.03428E11 cyprus CYP 2019 3.6 2.57583577745438E10 czech republic CZE 2019 12.2 2.52498032247163E11 denmark DNK 2019 10.7 3.47561349210979E11 djibouti DJI 2019 9.6 3.32461554909099E9 dominican republic DOM 2019 4.9 8.89412997335018E10 early-demographic dividend EAR 2019 8.47105731803596 1.15273363236526E13 east asia & pacific EAS 2019 7.89246535825769 2.69814555142245E13 east asia & pacific (ida & ibrd countries) TEA 2019 6.83672625050256 1.71136447545228E13 east asia & pacific (excluding high income) EAP 2019 6.86800122479052 1.71354045339684E13 ecuador ECU 2019 7.6 1.08108009E11 egypt EGY 2019 3.0 3.03080865603645E11 el salvador SLV 2019 6.1 2.689666E10 equatorial guinea GNQ 2019 7.9 1.14172780449859E10 estonia EST 2019 14.9 3.10455917534901E10 eswatini SWZ 2019 29.4 4.49567948096886E9 ethiopia ETH 2019 5.4 9.59125906281412E10 euro area EMU 2019 11.1471947291334 1.34135680658125E13 europe & central asia ECS 2019 12.8184155582707 2.29340048202416E13 europe & central asia (ida & ibrd countries) TEC 2019 14.6732206018185 4.15345315986614E12 europe & central asia (excluding high income) ECA 2019 14.9635207561178 3.49392638885403E12 european union EUU 2019 11.3427680473233 1.56893572473445E13 fiji FJI 2019 9.0 5.49626476637285E9 finland FIN 2019 15.3 2.6850820012549E11 fragile and conflict affected situations FCS 2019 5.23551477869694 1.66176581531745E12 france FRA 2019 13.8 2.72887024670588E12 gabon GAB 2019 8.4 1.68744058397948E10 gambia, the GMB 2019 4.8 1.81252910461923E9 georgia GEO 2019 9.2 1.74772555622583E10 germany DEU 2019 12.3 3.88832678862744E12 ghana GHA 2019 6.6 6.83375378157703E10 greece GRC 2019 5.1 2.05144152830847E11 grenada GRD 2019 0.7 1.21269440740741E9 guatemala GTM 2019 5.9 7.70200152013201E10 guinea GIN 2019 7.0 1.35138092582131E10 guinea-bissau GNB 2019

__Uticaj alkohola na sreću naroda u državi__

In [0]:
joinedDF = newHapDF.join(newAlcDF, newHapDF.country == newAlcDF.country, "inner").drop(newAlcDF.country).drop(newAlcDF.gdp).select("rank","country","score","both","male","female", "gdp")
joinedDF.display()
df = joinedDF.sort(joinedDF.score.desc())

rank country score both male female gdp 154 afghanistan 3.203 0.013 0.022 0.003 1.87994507427823E10 107 albania 4.719 6.82 11.02 2.53 1.54002428748812E10 88 algeria 5.211 0.63 1.09 0.17 1.7176740374819E11 47 argentina 6.086 9.45 14.66 4.64 4.51932356085842E11 116 armenia 4.559 4.67 7.7 2.14 1.36192913612814E10 11 australia 7.228 10.36 15.72 5.11 1.39195251037048E12 10 austria 7.246 11.93 18.42 5.76 4.4501187270447E11 90 azerbaijan 5.208 0.98 1.67 0.32 4.81742352941176E10 37 bahrain 6.199 1.13 1.54 0.32 3.8652579787234E10 125 bangladesh 4.456 0.0 0.0 0.0 3.0257132044575E11 81 belarus 5.323 10.96 17.93 5.13 6.44096471938044E10 18 belgium 6.923 10.8 16.7 5.13 5.35288715239999E11 102 benin 4.883 2.23 3.59 0.9 1.43916866327821E10 95 bhutan 5.082 0.21 0.33 0.077 2.53565706896575E9 61 bolivia 5.779 3.91 6.0 1.84 4.08953228437851E10 78 bosnia and herzegovina 5.386 7.81 12.95 2.96 2.02013232600733E10 148 botswana 3.488 6.61 11.1 2.61 1.65937206556402E10 32 brazil 6.3 7.32 11.65 3.22 1.87782427372078E12 97 bulgaria 5.011 12.46 19.86 5.59 6.89154161419576E10 115 burkina faso 4.587 11.05 18.5 3.85 1.61781620300694E10 145 burundi 3.775 7.45 12.22 2.89 2.63143436322693E9 109 cambodia 4.7 7.84 12.9 3.21 2.7089389786979E10 96 cameroon 5.044 5.52 8.76 2.32 3.96709773327348E10 9 canada 7.278 8.81 13.73 4.01 1.74201504548231E12 155 central african republic 3.083 1.71 2.76 0.71 2.22030736869593E9 132 chad 4.35 1.26 2.07 0.46 1.13149513427807E10 26 chile 6.444 8.95 13.74 4.36 2.79385487344492E11 93 china 5.191 6.04 9.37 2.6 1.4279937467431E13 43 colombia 6.125 5.45 8.84 2.27 3.23429888934257E11 142 comoros 3.973 1.15 2.0 0.3 1.19255973937288E9 12 costa rica 7.167 4.07 6.49 1.68 6.4072870931603E10 75 croatia 5.432 8.73 13.96 3.99 6.22462063405469E10 49 cyprus 6.046 10.8 16.92 4.76 2.57583577745438E10 2 denmark 7.6 10.13 15.58 4.8 3.47561349210979E11 77 dominican republic 5.425 6.68 10.62 2.8 8.89412997335018E10 50 ecuador 6.028 3.27 5.01 1.56 1.08108009E11 137 egypt 4.166 0.14 0.24 0.037 3.03080865603645E11 35 el salvador 6.253 4.09 6.91 1.75 2.689666E10 55 estonia 5.893 10.75 17.35 5.06 3.10455917534901E10 134 ethiopia 4.286 2.2 3.66 0.78 9.59125906281412E10 1 finland 7.769 10.65 16.56 4.99 2.6850820012549E11 24 france 6.592 12.23 19.21 5.83 2.72887024670588E12 104 gabon 4.799 8.08 12.24 3.7 1.68744058397948E10 119 georgia 4.519 9.47 15.61 4.11 1.74772555622583E10 17 germany 6.985 12.79 19.71 6.15 3.88832678862744E12 98 ghana 4.996 2.78 4.37 1.16 6.83375378157703E10 82 greece 5.287 10.5 16.72 4.6 2.05144152830847E11 27 guatemala 6.436 1.63 2.63 0.69 7.70200152013201E10 118 guinea 4.534 1.09 1.94 0.36 1.35138092582131E10 147 haiti 3.597 3.02 4.96 1.19 1.47858393829002E10 59 honduras 5.86 3.93 6.34 1.57 2.49155228491544E10 62 hungary 5.758 11.07 17.66 5.24 1.6352649143329E11 4 iceland 7.494 9.21 14.03 4.39 2.48577404450401E10 140 india 4.015 5.61 9.06 1.89 2.87050409671777E12 92 indonesia 5.192 0.22 0.37 0.063 1.11909125907462E12 117 iran 4.548 1.01 1.76 0.25 2.58245497664394E11 126 iraq 4.437 0.37 0.64 0.1 2.35097182233503E11 16 ireland 7.021 12.75 19.38 6.34 3.99122063504148E11 13 israel 7.139 4.38 6.84 2.02 3.9793459695256E11 36 italy 6.223 8.01 12.74 3.61 2.00938386730745E12 56 jamaica 5.89 4.16 6.71 1.69 1.58307685498916E10 58 japan 5.886 10.09 15.58 4.93 5.14878194847817E12 101 jordan 4.906 0.52 0.89 0.14 4.4502895915493E10 60 kazakhstan 5.809 5.0 8.33 2.02 1.81667190075541E11 121 kenya 4.509 2.15 3.51 0.83 1.00555485831944E11 51 kuwait 6.021 0.0 0.0 0.0 1.36196760210804E11 53 latvia 5.94 13.19 21.61 6.31 3.43087838253019E10 91 lebanon 5.197 1.53 2.58 0.49 5.19537445074091E10 144 lesotho 3.802 5.08 8.65 1.67 2.36621306857507E9 141 liberia 3.975 5.37 8.58 2.2 3.3195965E9 72 libya 5.525 0.027 0.047 0.007 5.20911522283425E10 42 lithuania 6.149 12.78 20.6 6.3 5.46973790173332E10 14 luxembourg 7.09 12.45 18.74 6.08 7.01957154955136E10 143 madagascar 3.933 1.98 3.22 0.77 1.410466451487E10 150 malawi 3.41 4.08 6.76 1.56 1.1025357837607E10 80 malaysia

In [0]:

df.createOrReplaceTempView("df_table")
print(df.head())
print(df.tail(1))

Row(rank=1, country='finland', score=7.769, both=10.65, male=16.56, female=4.99, gdp=268508200125.49)
[Row(rank=155, country='central african republic', score=3.083, both=1.71, male=2.76, female=0.71, gdp=2220307368.69593)]


Iz gornjih podataka možemo videti da Finska ima najveci koeficijent sreće 7.769, a Centralnoafrička republika ima najmanji koeficijent sreće 3.083. Takođe možemo primetiti da Finska konzumira 10.65 litra čistog alkohola 

godišnje, što je znazno veće od Centralnoafričke republike koja konzumira 1.71 litra čistog alkogola godišnje. Ovo može da znači da je Finska srećnija jer konzumira više alkohola, ali može biti i do razlike u ekonomskoj

razvijenosti zemlje. Za sledeći primer uzećemo zemlje sa sličnim ekonomskim stanjem.

In [0]:
%sql
SELECT country, score, both, gdp from df_table where country == "france" OR country == "italy"

country score both gdp france 6.592 12.23 2.72887024670588E12 italy 6.223 8.01 2.00938386730745E12

U gornjem primeru vidimo da Francuska konzumira više alkohola i da je koeficijent sreće veći u odnosu na Italiju

In [0]:
%sql
SELECT country, score, both, gdp from df_table where country == "germany" OR country == "austria"

country score both gdp austria 7.246 11.93 4.4501187270447E11 germany 6.985 12.79 3.88832678862744E12

U gornjem primeru vidimo da iako Nemačka konzumira više alkohola, nema veći koeficijent sreće. Ovo može da znaći da alkohol nema uticaj na koeficijent sreće države, ali isto moramo uzeti u obzir da je nacionalno piće Nemačke pilsner što je pivo, a nacionalno piće Austrije je almdudler što je tip soka.

In [0]:
%sql
SELECT country, score, both, gdp from df_table where country == "latvia" OR country == "turkey"

country score both gdp latvia 5.94 13.19 3.43087838253019E10 turkey 5.373 1.77 7.61004425605414E11

Za poslednji primer ćemo uzeti Latviu i Tursku. Latvija je zemlja sa najvišom konzumpcijom alkohola, a Turska je zemlja sa veoma niskom konzumpcijom alkohola. Razlika u konzumpciji alkohola je ogromna u odnosu na razliku u 

koeficijentu sreće.

__Rezultat__

U većinu primera vidimo korelaciju izmedju konzumpcije alkohola i većeg koeficijenta sreće u državi.

__Da li srećnije zemlje nužno imaju manju stopu samoubistava__

In [0]:
joinedDF = hapDF.join(suDF, hapDF.country == suDF.country, "inner").drop(suDF.country).select("rank", "country","score","suicide_rate", "gdp")
display(joinedDF)

rank country score suicide_rate gdp 154 afghanistan 3.203 4.1 1.87994507427823E10 107 albania 4.719 4.3 1.54002428748812E10 88 algeria 5.211 2.5 1.7176740374819E11 47 argentina 6.086 8.4 4.51932356085842E11 116 armenia 4.559 3.3 1.36192913612814E10 11 australia 7.228 12.5 1.39195251037048E12 10 austria 7.246 14.6 4.4501187270447E11 90 azerbaijan 5.208 4.1 4.81742352941176E10 37 bahrain 6.199 8.9 3.8652579787234E10 125 bangladesh 4.456 3.7 3.0257132044575E11 81 belarus 5.323 21.2 6.44096471938044E10 18 belgium 6.923 18.3 5.35288715239999E11 102 benin 4.883 7.8 1.43916866327821E10 95 bhutan 5.082 4.6 2.53565706896575E9 61 bolivia 5.779 6.2 4.08953228437851E10 78 bosnia and herzegovina 5.386 10.9 2.02013232600733E10 148 botswana 3.488 16.1 1.65937206556402E10 32 brazil 6.3 6.9 1.87782427372078E12 97 bulgaria 5.011 9.7 6.89154161419576E10 115 burkina faso 4.587 7.5 1.61781620300694E10 145 burundi 3.775 6.2 2.63143436322693E9 109 cambodia 4.7 4.9 2.7089389786979E10 96 cameroon 5.044 9.0 3.96709773327348E10 9 canada 7.278 11.8 1.74201504548231E12 155 central african republic 3.083 12.3 2.22030736869593E9 132 chad 4.35 6.4 1.13149513427807E10 26 chile 6.444 9.0 2.79385487344492E11 93 china 5.191 8.1 1.4279937467431E13 43 colombia 6.125 3.9 3.23429888934257E11 142 comoros 3.973 5.4 1.19255973937288E9 127 congo, dem. rep. 4.418 6.7 5.0400747050496E10 103 congo, rep. 4.812 6.5 1.27503390229512E10 12 costa rica 7.167 8.1 6.4072870931603E10 75 croatia 5.432 16.4 6.22462063405469E10 49 cyprus 6.046 3.6 2.57583577745438E10 20 czech republic 6.852 12.2 2.52498032247163E11 2 denmark 7.6 10.7 3.47561349210979E11 77 dominican republic 5.425 4.9 8.89412997335018E10 50 ecuador 6.028 7.6 1.08108009E11 137 egypt 4.166 3.0 3.03080865603645E11 35 el salvador 6.253 6.1 2.689666E10 55 estonia 5.893 14.9 3.10455917534901E10 134 ethiopia 4.286 5.4 9.59125906281412E10 1 finland 7.769 15.3 2.6850820012549E11 24 france 6.592 13.8 2.72887024670588E12 104 gabon 4.799 8.4 1.68744058397948E10 119 georgia 4.519 9.2 1.74772555622583E10 17 germany 6.985 12.3 3.88832678862744E12 98 ghana 4.996 6.6 6.83375378157703E10 82 greece 5.287 5.1 2.05144152830847E11 27 guatemala 6.436 5.9 7.70200152013201E10 118 guinea 4.534 7.0 1.35138092582131E10 147 haiti 3.597 9.6 1.47858393829002E10 59 honduras 5.86 2.1 2.49155228491544E10 62 hungary 5.758 16.6 1.6352649143329E11 4 iceland 7.494 11.9 2.48577404450401E10 140 india 4.015 12.7 2.87050409671777E12 92 indonesia 5.192 2.4 1.11909125907462E12 117 iran 4.548 5.2 2.58245497664394E11 126 iraq 4.437 3.6 2.35097182233503E11 16 ireland 7.021 9.6 3.99122063504148E11 13 israel 7.139 5.3 3.9793459695256E11 36 italy 6.223 6.7 2.00938386730745E12 56 jamaica 5.89 2.4 1.58307685498916E10 58 japan 5.886 15.3 5.14878194847817E12 101 jordan 4.906 1.6 4.4502895915493E10 60 kazakhstan 5.809 17.6 1.81667190075541E11 121 kenya 4.509 6.1 1.00555485831944E11 51 kuwait 6.021 2.9 1.36196760210804E11 53 latvia 5.94 20.1 3.43087838253019E10 91 lebanon 5.197 2.8 5.19537445074091E10 144 lesotho 3.802 72.4 2.36621306857507E9 141 liberia 3.975 4.5 3.3195965E9 72 libya 5.525 4.5 5.20911522283425E10 42 lithuania 6.149 26.1 5.46973790173332E10 14 luxembourg 7.09 11.3 7.01957154955136E10 143 madagascar 3.933 5.5 1.410466451487E10 150 malawi 3.41 5.4 1.1025357837607E10 80 malaysia 5.339 5.7 3.65276282438141E11 128 mali 4.39 4.1 1.72802511939527E10 22 malta 6.726 6.1 1.52157143090974E10 122 mauritania 4.49 3.1 7.88966245216785E9 57 mauritius 5.888 9.5 1.4045808843221E10 23 mexico 6.595 5.3 1.26943393280591E12 71 moldova 5.529 14.7 1.19702339398808E10 83 mongolia 5.285 17.9 1.42063590068095E10 73 montenegro 5.523 21.0 5.54267431706225E9 89 morocco 5.208 7.2 1.19870439113662E11 123 mozambique 4.466 13.6 1.53847017979863E10 131 myanmar 4.36 2.9 6.86977614771818E10 113 namibia 4.639 9.7 1.24967817197752E10 100 nepal 4.913 9.0 3.41861806949285E10 5 netherlands 7.488 11.8 9.10194347568626E11 8 new zealand 7.307 11.0 2.12891048340608E11 45 nicaragua 6.105 4.4 1.2611218

U gornjem primeru možemo da vidimo da zemlje sa višim koeficijentom sreće mogu imati skoro duplo veću stopu samoubistva od zemlja sa nižim koeficijentom sreće

In [0]:
joinedDF = joinedDF.sort(joinedDF.score.desc())
df = joinedDF
df.createOrReplaceTempView("df_table")

In [0]:
display(df.take(10))
display(df.tail(10))



rank country score suicide_rate gdp 1 finland 7.769 15.3 2.6850820012549E11 2 denmark 7.6 10.7 3.47561349210979E11 3 norway 7.554 11.8 4.04941363636364E11 4 iceland 7.494 11.9 2.48577404450401E10 5 netherlands 7.488 11.8 9.10194347568626E11 6 switzerland 7.48 14.5 7.31767398052879E11 7 sweden 7.343 14.7 5.33879529188454E11 8 new zealand 7.307 11.0 2.12891048340608E11 9 canada 7.278 11.8 1.74201504548231E12 10 austria 7.246 14.6 4.4501187270447E11

rank country score suicide_rate gdp 144 lesotho 3.802 72.4 2.36621306857507E9 145 burundi 3.775 6.2 2.63143436322693E9 146 zimbabwe 3.663 14.1 1.92842897390517E10 147 haiti 3.597 9.6 1.47858393829002E10 148 botswana 3.488 16.1 1.65937206556402E10 150 malawi 3.41 5.4 1.1025357837607E10 152 rwanda 3.334 5.6 1.03559742173735E10 153 tanzania 3.231 4.3 6.11368736923985E10 154 afghanistan 3.203 4.1 1.87994507427823E10 155 central african republic 3.083 12.3 2.22030736869593E9

U gornjem primeru imamo 10 zemlja sa najvećim koeficijentom sreće i 10 zemlja sa najnižim koeficijentoim sreće. Možemo primetiti da FInska koja ima najveći koeficijent sreće ima skoro 4 puta veću stopu samoubistva od Afganistana koji ima duplo manji koeficijent sreće

In [0]:
df = df.sort(df.suicide_rate.desc())
display(df.take(10))
display(df.tail(10))

rank country score suicide_rate gdp 144 lesotho 3.802 72.4 2.36621306857507E9 42 lithuania 6.149 26.1 5.46973790173332E10 68 russia 5.648 25.1 1.68744852546661E12 106 south africa 4.722 23.5 3.8793457409817E11 133 ukraine 4.332 21.6 1.53882982016281E11 33 uruguay 6.293 21.2 6.12311498805857E10 81 belarus 5.323 21.2 6.44096471938044E10 73 montenegro 5.523 21.0 5.54267431706225E9 53 latvia 5.94 20.1 3.43087838253019E10 44 slovenia 6.118 19.8 5.41788776059999E10

rank country score suicide_rate gdp 31 panama 6.321 2.9 6.6984427150321E10 91 lebanon 5.197 2.8 5.19537445074091E10 65 peru 5.697 2.8 2.28470919605669E11 88 algeria 5.211 2.5 1.7176740374819E11 92 indonesia 5.192 2.4 1.11909125907462E12 56 jamaica 5.89 2.4 1.58307685498916E10 79 turkey 5.373 2.4 7.61004425605414E11 69 philippines 5.631 2.2 3.76823278560849E11 59 honduras 5.86 2.1 2.49155228491544E10 101 jordan 4.906 1.6 4.4502895915493E10

U gornjem primeru imamo 10 zemlja sa najvećom stopom samoubistva i 10 zemlja sa najnižom stopom samoubistva. U prvom grafu se većinski nalaze zemlje kategarizovane u zemlje drugog sveta, pored Lesata koji je

zemlja trećeg sveta i Slovenije koja može da se smatra zemljom prvog sveta. U donjem grafu su sve zemlje trećeg sveta. Zdravorazumski zar ne bi trebalo da razvijenije zemlje imaju veći koeficijent sreće i manju stopu samoubistva

ali ovde vidimo da to nije uvek istina.

__Rezultat__

Veći koeficijent sreće ne znači da će država da ima manju stopu samoubistva.

__Uticaj konzumacije alkohola na stopu samoubistva__

In [0]:
#sortiramo podatke po stopi samoubistva opadajuće
df = newAlcDF.sort(newAlcDF.suicide_rate.desc())
df = df.where((df.both != 0) & (df.male != 0) & (df.female != 0))
df.display()
df.createOrReplaceTempView("df_table")

country both male female iso3c suicide_rate gdp lesotho 5.08 8.65 1.67 LSO 72.4 2.36621306857507E9 guyana 5.33 8.57 2.1 GUY 40.3 5.17376019184652E9 eswatini 8.78 14.78 3.26 SWZ 29.4 4.49567948096886E9 kiribati 2.32 3.97 0.8 KIR 28.3 1.88391770636764E8 lithuania 12.78 20.6 6.3 LTU 26.1 5.46973790173332E10 suriname 7.4 11.74 3.1 SUR 25.4 3.98448376237123E9 russia 10.5 17.99 4.31 RUS 25.1 1.68744852546661E12 south africa 9.45 15.74 3.46 ZAF 23.5 3.8793457409817E11 ukraine 8.34 14.0 3.65 UKR 21.6 1.53882982016281E11 belarus 10.96 17.93 5.13 BLR 21.2 6.44096471938044E10 uruguay 6.92 10.82 3.38 URY 21.2 6.12311498805857E10 montenegro 12.24 19.51 5.3 MNE 21.0 5.54267431706225E9 latvia 13.19 21.61 6.31 LVA 20.1 3.43087838253019E10 slovenia 12.11 18.68 5.67 SVN 19.8 5.41788776059999E10 belgium 10.8 16.7 5.13 BEL 18.3 5.35288715239999E11 vanuatu 2.1 3.48 0.73 VUT 18.0 9.34521604602009E8 mongolia 5.9 9.23 2.72 MNG 17.9 1.42063590068095E10 kazakhstan 5.0 8.33 2.02 KAZ 17.6 1.81667190075541E11 hungary 11.07 17.66 5.24 HUN 16.6 1.6352649143329E11 croatia 8.73 13.96 3.99 HRV 16.4 6.22462063405469E10 united states 9.97 15.44 4.69 USA 16.1 2.1433224697E13 botswana 6.61 11.1 2.61 BWA 16.1 1.65937206556402E10 finland 10.65 16.56 4.99 FIN 15.3 2.6850820012549E11 japan 10.09 15.58 4.93 JPN 15.3 5.14878194847817E12 estonia 10.75 17.35 5.06 EST 14.9 3.10455917534901E10 sweden 9.04 13.85 4.27 SWE 14.7 5.33879529188454E11 moldova 12.85 20.61 5.89 MDA 14.7 1.19702339398808E10 solomon islands 1.74 2.87 0.59 SLB 14.7 1.57009322925588E9 austria 11.93 18.42 5.76 AUT 14.6 4.4501187270447E11 switzerland 11.23 17.18 5.45 CHE 14.5 7.31767398052879E11 cuba 6.26 9.97 2.64 CUB 14.5 1.03428E11 zimbabwe 4.52 7.94 1.62 ZWE 14.1 1.92842897390517E10 sri lanka 2.87 4.75 1.19 LKA 14.0 8.39909469394651E10 france 12.23 19.21 5.83 FRA 13.8 2.72887024670588E12 mozambique 2.69 4.53 1.04 MOZ 13.6 1.53847017979863E10 india 5.61 9.06 1.89 IND 12.7 2.87050409671777E12 samoa 2.82 4.58 0.93 WSM 12.6 8.52007112724637E8 australia 10.36 15.72 5.11 AUS 12.5 1.39195251037048E12 central african republic 1.71 2.76 0.71 CAF 12.3 2.22030736869593E9 germany 12.79 19.71 6.15 DEU 12.3 3.88832678862744E12 slovakia 11.06 17.26 5.3 SVK 12.1 1.05284375640698E11 iceland 9.21 14.03 4.39 ISL 11.9 2.48577404450401E10 netherlands 9.67 14.9 4.54 NLD 11.8 9.10194347568626E11 canada 8.81 13.73 4.01 CAN 11.8 1.74201504548231E12 norway 7.14 10.8 3.45 NOR 11.8 4.04941363636364E11 portugal 12.09 19.51 5.59 PRT 11.5 2.39986922638902E11 serbia 8.85 14.09 3.93 SRB 11.4 5.15142223818428E10 poland 11.89 18.65 5.65 POL 11.3 5.97280564671563E11 luxembourg 12.45 18.74 6.08 LUX 11.3 7.01957154955136E10 singapore 2.0 2.92 0.99 SGP 11.2 3.74386306993109E11 new zealand 10.69 16.46 5.23 NZL 11.0 2.12891048340608E11 bosnia and herzegovina 7.81 12.95 2.96 BIH 10.9 2.02013232600733E10 denmark 10.13 15.58 4.8 DNK 10.7 3.47561349210979E11 bulgaria 12.46 19.86 5.59 BGR 9.7 6.89154161419576E10 namibia 3.15 5.3 1.2 NAM 9.7 1.24967817197752E10 romania 12.34 19.5 5.68 ROU 9.7 2.49881592298072E11 ireland 12.75 19.38 6.34 IRL 9.6 3.99122063504148E11 djibouti 0.36 0.6 0.093 DJI 9.6 3.32461554909099E9 haiti 3.02 4.96 1.19 HTI 9.6 1.47858393829002E10 mauritius 4.78 7.67 2.0 MUS 9.5 1.4045808843221E10 north macedonia 6.43 10.36 2.54 MKD 9.4 1.2606338448547E10 georgia 9.47 15.61 4.11 GEO 9.2 1.74772555622583E10 nepal 0.62 1.1 0.25 NPL 9.0 3.41861806949285E10 fiji 3.68 6.08 1.23 FJI 9.0 5.49626476637285E9 cameroon 5.52 8.76 2.32 CMR 9.0 3.96709773327348E10 chile 8.95 13.74 4.36 CHL 9.0 2.79385487344492E11 pakistan 0.31 0.52 0.095 PAK 8.9 2.79056608888638E11 bahrain 1.13 1.54 0.32 BHR 8.9 3.8652579787234E10 thailand 8.5 13.75 3.61 THA 8.8 5.44263840039166E11 togo 2.66 4.25 1.1 TGO 8.8 7.2203952477424E9 trinidad and tobago 6.52 10.24 2.94 TTO 8.7 2.32083265475327E10 argentina 9.45 14.66 4.64 ARG 8.4 4.51932356085842E11 gabon 8.08 12.24 3.7 GAB 8.4 1.68744058397948E10 costa rica 4.07 6.49 1.68 CRI 8.1 6.4072870931603E10 china 6.04 9.37 2.6 C

Na gornjem primeru možemo videti da u nekim slučajevima sa većom konzumpcijom alkohola raste i stopa samoubistva

In [0]:
%sql
SELECT country from df_table where female > male



country

Možemo primetiti da ne postoji država gde žene imaju veću konzumpciju alkohola od muškarca

In [0]:
display(df.take(15))
display(df.tail(15))

country both male female iso3c suicide_rate gdp lesotho 5.08 8.65 1.67 LSO 72.4 2.36621306857507E9 guyana 5.33 8.57 2.1 GUY 40.3 5.17376019184652E9 eswatini 8.78 14.78 3.26 SWZ 29.4 4.49567948096886E9 kiribati 2.32 3.97 0.8 KIR 28.3 1.88391770636764E8 lithuania 12.78 20.6 6.3 LTU 26.1 5.46973790173332E10 suriname 7.4 11.74 3.1 SUR 25.4 3.98448376237123E9 russia 10.5 17.99 4.31 RUS 25.1 1.68744852546661E12 south africa 9.45 15.74 3.46 ZAF 23.5 3.8793457409817E11 ukraine 8.34 14.0 3.65 UKR 21.6 1.53882982016281E11 uruguay 6.92 10.82 3.38 URY 21.2 6.12311498805857E10 belarus 10.96 17.93 5.13 BLR 21.2 6.44096471938044E10 montenegro 12.24 19.51 5.3 MNE 21.0 5.54267431706225E9 latvia 13.19 21.61 6.31 LVA 20.1 3.43087838253019E10 slovenia 12.11 18.68 5.67 SVN 19.8 5.41788776059999E10 belgium 10.8 16.7 5.13 BEL 18.3 5.35288715239999E11

country both male female iso3c suicide_rate gdp panama 7.77 12.23 3.36 PAN 2.9 6.6984427150321E10 lebanon 1.53 2.58 0.49 LBN 2.8 5.19537445074091E10 peru 6.78 10.41 3.24 PER 2.8 2.28470919605669E11 maldives 2.78 3.95 0.57 MDV 2.7 5.60776262514627E9 algeria 0.63 1.09 0.17 DZA 2.5 1.7176740374819E11 indonesia 0.22 0.37 0.063 IDN 2.4 1.11909125907462E12 jamaica 4.16 6.71 1.69 JAM 2.4 1.58307685498916E10 turkey 1.77 3.12 0.48 TUR 2.4 7.61004425605414E11 philippines 7.02 11.22 2.87 PHL 2.2 3.76823278560849E11 honduras 3.93 6.34 1.57 HND 2.1 2.49155228491544E10 jordan 0.52 0.89 0.14 JOR 1.6 4.4502895915493E10 sao tome and principe 5.8 9.15 2.49 STP 1.5 4.27425039684339E8 grenada 8.96 14.25 3.64 GRD 0.7 1.21269440740741E9 barbados 10.37 16.95 4.35 BRB 0.6 5.209E9 antigua and barbuda 9.4 15.09 4.25 ATG 0.4 1.68753333333333E9

Na gornjim primerima možemo videti da u većini primera sa manjom konzumpcijom alkohola imamo i manju stopu samoubistva

In [0]:
#Sortiramo podatke po konzumpciji alkohola opadajuće
df = df.sort(df.both.desc())
display(df.take(15))
display(df.tail(15))

country both male female iso3c suicide_rate gdp latvia 13.19 21.61 6.31 LVA 20.1 3.43087838253019E10 moldova 12.85 20.61 5.89 MDA 14.7 1.19702339398808E10 germany 12.79 19.71 6.15 DEU 12.3 3.88832678862744E12 lithuania 12.78 20.6 6.3 LTU 26.1 5.46973790173332E10 ireland 12.75 19.38 6.34 IRL 9.6 3.99122063504148E11 spain 12.67 19.94 5.76 ESP 7.7 1.39304609313725E12 uganda 12.48 20.61 4.94 UGA 4.6 3.53530606342022E10 bulgaria 12.46 19.86 5.59 BGR 9.7 6.89154161419576E10 luxembourg 12.45 18.74 6.08 LUX 11.3 7.01957154955136E10 romania 12.34 19.5 5.68 ROU 9.7 2.49881592298072E11 montenegro 12.24 19.51 5.3 MNE 21.0 5.54267431706225E9 france 12.23 19.21 5.83 FRA 13.8 2.72887024670588E12 slovenia 12.11 18.68 5.67 SVN 19.8 5.41788776059999E10 portugal 12.09 19.51 5.59 PRT 11.5 2.39986922638902E11 tanzania 12.04 19.89 4.36 TZA 4.3 6.11368736923985E10

country both male female iso3c suicide_rate gdp algeria 0.63 1.09 0.17 DZA 2.5 1.7176740374819E11 nepal 0.62 1.1 0.25 NPL 9.0 3.41861806949285E10 timor-leste 0.53 0.84 0.21 TLS 3.7 2.0479317E9 jordan 0.52 0.89 0.14 JOR 1.6 4.4502895915493E10 niger 0.52 0.89 0.15 NER 5.3 1.29164551611081E10 morocco 0.49 0.87 0.13 MAR 7.2 1.19870439113662E11 tonga 0.43 0.73 0.15 TON 3.8 5.12350059421629E8 iraq 0.37 0.64 0.1 IRQ 3.6 2.35097182233503E11 djibouti 0.36 0.6 0.093 DJI 9.6 3.32461554909099E9 pakistan 0.31 0.52 0.095 PAK 8.9 2.79056608888638E11 indonesia 0.22 0.37 0.063 IDN 2.4 1.11909125907462E12 bhutan 0.21 0.33 0.077 BTN 4.6 2.53565706896575E9 egypt 0.14 0.24 0.037 EGY 3.0 3.03080865603645E11 libya 0.027 0.047 0.007 LBY 4.5 5.20911522283425E10 afghanistan 0.013 0.022 0.003 AFG 4.1 1.87994507427823E10

Kao i na prethodnom primeru možemo videti da u većini slučajeva manja konzumpcija alkohola znači i manja stopa samoubistva

__Rezultat__

Konzumpcija alkohola može da utiče na stopu samoubistva u državi ali ne u prevelikoj meri.

__Zaključak__

Ako uzmemo rezultate hipoteze iz ove analize dobićemo da države koje imaju veću konzumpciju alkohola su srećnije, da zemlje koje imaju veći koeficijent sreće nemaju nužno manju stopu stopu samoubistva, ponekad imaju i veću od država koje imaju manji koeficijent sreće i da konzumpcija alkohola povećava stopu samoubistva. Ovo znači da ovaj način merenja sreće nije adekvatan ili da srećni ljudi izvršavaju samoubistvo i da više konzumiraju alkohol.